Formula/ecl.rb

```diff
@@ -1,12 +1,20 @@
 class Ecl < Formula
   desc "Embeddable Common Lisp"
   homepage "https://common-lisp.net/project/ecl/"
-  url "https://common-lisp.net/project/ecl/static/files/release/ecl-21.2.1.tgz"
-  sha256 "b15a75dcf84b8f62e68720ccab1393f9611c078fcd3afdd639a1086cad010900"
   license "LGPL-2.1-or-later"
-  revision 1
+  revision 2
   head "https://gitlab.com/embeddable-common-lisp/ecl.git", branch: "develop"
 
+  stable do
+    url "https://common-lisp.net/project/ecl/static/files/release/ecl-21.2.1.tgz"
+    sha256 "b15a75dcf84b8f62e68720ccab1393f9611c078fcd3afdd639a1086cad010900"
+
+    # Backport fix for bug that causes errors when building `sbcl`.
+    # Issue ref: https://gitlab.com/embeddable-common-lisp/ecl/-/issues/667
+    # Remove in the next release along with the stable block
+    patch :DATA
```

Comment:
Noting that backport is needed since upstream moved some of code from `cmploc.lsp` to `cmppass2-loc.lsp` so patch doesn't apply cleanly.

https://gitlab.com/embeddable-common-lisp/ecl/-/commit/f3d4cf4b66ab6c3cd8629ab6d0c7f7c50d7fd8a4#a43e5089d1e0274b42fe8a2bcf50b7ea7c2f12f1